clang/lib/AST/TemplateName.cpp

```diff
@@ -292,6 +292,14 @@ void TemplateName::Profile(llvm::FoldingSetNodeID &ID) {
 
 void TemplateName::print(raw_ostream &OS, const PrintingPolicy &Policy,
                          Qualified Qual) const {
+  auto handleCanonicalTTP = [](TemplateDecl *TD, raw_ostream &OS) {
```

Comment:
Shouldn't that be "anonymous" rather than canonical (even if canonical implies anonymous)

Reply:
Yeah, the primary use case is canonical TTPs, but it would be more accurate to say this handles any anonymous TTP, for whatever reason.